### DISCUSSION TITLE
`litera` navbar background color used to be `$white`, now is `$primary`

### DISCUSSION THREAD

**Discussion by DavisVaughan** [quarto-dev/quarto-cli#10973] [Q&A] [answered]
### Bug description

My website is here https://blog.davisvaughan.com/

It uses `litera` as a theme
https://github.com/DavisVaughan/blog/blob/a97cc9ff32895ff1d05f0fe4ccb6490b2352d452/custom.scss#L4

It has always had a white navbar background

![Image](https://github.com/user-attachments/assets/2c8988a1-7205-439e-921e-feb736e999f1)

I just updated Quarto locally after a looooong time, and now the primary color is being used as the navbar background (im not setting `$primary` anywhere).

![Image](https://github.com/user-attachments/assets/daad9ce7-d1b7-4ab7-ab79-3632b93cbcec)

I greatly prefer the white default.

I've tracked this back to this bug fix https://github.com/quarto-dev/quarto-cli/issues/7447

Corresponding to this commit https://github.com/quarto-dev/quarto-cli/commit/ef91ac8330e8d76a3c8de33008d77fb82ec9d9f9

Particularly:

![Image](https://github.com/user-attachments/assets/24626cd1-11f5-4187-b4a2-4417073d8c1c)

So it looks like it definitely used to just be `$white`.

Was that change intended? I can fix it with this, but I just wanted to make sure this was purposeful.

```
website:
  navbar:
    background: white
```

The version this change was introduced in is v1.4.479.

### Quarto check output

quarto v1.4.479 is the first version this happened in

**Comment by cscheid** [comment]
Let's move this to a discussion - it was clearly an intentional choice.

**Comment by DavisVaughan** [comment, marked as answer]
If I pull up "litera theme" on the Bootswatch website, the first thing it shows you is that blue navbar. Maybe it was supposed to do that all along? https://bootswatch.com/litera/

**Reply by cscheid** [reply]
~~Yes, I think that's the case.~~

See below, though: I can't repro your report on a clean project.

**Comment by cderv** [comment]
Personally, I think there is more to it when I see 
https://github.com/quarto-dev/quarto-cli/blob/5c38d77c240f3eb2a6ebf5239a31cf7c7ea9be07/src/resources/formats/html/bootstrap/_bootstrap-functions.scss#L138-L140

`litera` is part of the themes where quarto initially choose some other defaults than Boostwatch. (this is what `theme_override` is about). And the change from https://github.com/quarto-dev/quarto-cli/commit/ef91ac8330e8d76a3c8de33008d77fb82ec9d9f9 are supposed to keep that for default, but do respect other value for `$primary`. And to me this is what https://github.com/quarto-dev/quarto-cli/issues/7447 was about. 

So I would like to understand more on this as it seems to me ef91ac8330e8d76a3c8de33008d77fb82ec9d9f9  to fix #7447 is not completely working in the sense the initial quarto's default are not kept. 

Though, maybe we may prefer this new defaults now. But I don't understand ef91ac8330e8d76a3c8de33008d77fb82ec9d9f9 change in that case. 

To be discussed.

**Reply by DavisVaughan** [reply]
Happy to test anything if you decide to make a change in some way here, just ping me!

**Reply by cscheid** [reply]
Christophe is right, but ugh, I honestly hate all of that code. This SCSS overrides code we have is annoying to test, hard to explain to users, and easy to regress on. That's a recipe for this kind of problem.

I wish we would err towards respecting the bootswatch theme, and then allow users to override as they choose to.

**Reply by cscheid** [reply]
@DavisVaughan, on a plain website created by Quarto, I _do_ get a white navbar:

```
$ quarto create project website
...
```

Edit `_quarto.yml` to look like this:

```
project:
  type: website

website:
  title: "test-navbar-scss"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - about.qmd

format:
  html:
    theme: litera
    css: styles.css
    toc: true
```

Then render a page. I see this:

<img width="854" alt="image" src="https://github.com/user-attachments/assets/18099e6f-b9ec-4a79-8df4-91ac1902e189">

**Reply by mcanouil** [reply]
> Christophe is right, but ugh, I honestly hate all of that code. This SCSS overrides code we have is annoying to test, hard to explain to users, and easy to regress on. That's a recipe for this kind of problem.
> 
> I wish we would err towards respecting the bootswatch theme, and then allow users to override as they choose to.

Such as: (I did not have the time to get back on it)

- https://github.com/quarto-dev/quarto-cli/pull/10029

**Reply by DavisVaughan** [reply]
@cscheid you're right! Here's a reprex to reproduce what I see

```
quarto create project website
```

Edit `_quarto.yml` to look like

```
project:
  type: website

website:
  title: "test-navbar-scss"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - about.qmd

format:
  html:
    theme: custom.scss
    css: styles.css
```

Add `custom.scss` with the following

```
/*-- scss:defaults --*/

// Use litera as the base
$theme: "litera" !default;
```

Now render and I see this

<img width="1762" alt="Screenshot 2024-10-04 at 1 38 32 PM" src="https://github.com/user-attachments/assets/490255e4-bfef-4bb9-b2a8-660817d2dc5b">

**Reply by DavisVaughan** [reply]
Oh yea this is fun

https://github.com/user-attachments/assets/78a2ad95-b56d-46e5-81e6-2f74e15c9280

**Reply by cscheid** [reply]
I... don't think we advertise using the `theme` scss variable as the way to get the Litera theme?

**Reply by cscheid** [reply]
Look at the changed fonts. With your `custom.scss`, you're not actually getting the Litera theme. You're only getting some of the overrides. [Litera](https://bootswatch.com/litera/) uses Georgia for body type.

Funnily enough, the override you like is _not_ part of Litera, but _is_ Quarto. You just need to actually select the full Litera theme. Otherwise, that conditional Charles added in the commit flagged by Christophe is checking against an unexpected default value.

Overall, I stand by my point that the whole thing is too hard to reason about. If you can't get it right, our typical user stands no chance :(

**Reply by DavisVaughan** [reply]
I use litera plus some customizations. How else would I do that if `theme:` in `_quarto.yml` can only take either `litera` or `custom.scss`? https://github.com/DavisVaughan/blog/blob/main/custom.scss

**Reply by cscheid** [reply]
FWIW, if you're willing to use the dev version of Quarto, we now have tooling to help explain what happens in SCSS. You use it like this:

```
$ QUARTO_SAVE_SCSS=out quarto render index.qmd
...
$ ls out*.scss
out-1.scss out-2.scss [this happens because we generate two SCSS files]
$ cd ~/quarto-cli/tools/sass-variable-explainer
$ quarto preview sass-ui.qmd
```

You'll see a page that looks like this:

<img width="1023" alt="image" src="https://github.com/user-attachments/assets/a9eb85ae-572a-4b45-bcc5-829862c5c3ea">

Now open the correct out-* sass file (it'll be the largest one; the other one is only used for syntax highlighting)

You'll see all color variables in the SCSS file, and you can click on them to see the dependency chains.

**Reply by cscheid** [reply]
Like so:

https://github.com/user-attachments/assets/fad582e4-804a-46b9-88c7-a1a6218045b2

**Reply by gadenbuie** [reply]
> How else would I do that if `theme:` in `_quarto.yml` can only take either `litera` or `custom.scss`?

@DavisVaughan `theme` can [take an array](https://quarto.org/docs/output-formats/html-themes.html#customizing-themes), e.g. the base `litera` theme with additional `custom.scss` layered on top:

```yaml
theme:
  - litera
  - custom.scss
```

**Reply by cderv** [reply]
Other doc link about this is this one: https://quarto.org/docs/output-formats/html-themes.html#theme-options

And I believe our examples websites are also using this. 

I am curious how you ended using `$theme` variable for setting a quarto theme. Was it showed as example somewhere ? 

Or maybe the [custom theme](https://quarto.org/docs/output-formats/html-themes.html#custom-themes) part is not detailed enough as we link to theme's source code. 🤔
